src/components/AR/cameras/AR-SLAM-camera.ts

```diff
@@ -0,0 +1,80 @@
+import { ARSession } from '../AR-session';
+
+import { WorldTracking_XR8 } from '../frameworks/xr8/world-tracking-mode-xr8';
+import { WorldTracking_webAR } from '../frameworks/webAR/world-tracking-mode-webAR';
+
+
+import { ITrackingMode } from '../frameworks/trackingMode';
+import { webXRProvider } from '../frameworks/webAR/webXR-provider';
+import { xr8Provider } from '../frameworks/xr8/xr8-provider';
+import { ARCamera } from './AR-Camera';
+import { Type } from '@wonderlandengine/api';
+
+// running on a browser?
+if (window.document) {
+
+  // WL.arSupported might not exist at this point. so we wait until WLE resolves it
+  (function checkARSupport() {
```

Comment:
I think this code should be in the component itself and shouldn't be top level.

Reply:
@DavidPeicho 
this means that the whole 8th Wall lib will start loading only when the engine is fully loaded and initialized.  
What about we still keep the logic here outside the component, but check `navigator.xr.isSessionSupported('immersive-ar')` manually?

Reply:
I would then create a function or something that is required in the `index.js` file:

```js
export async function loadArSession() { ... }
```

Then  the user would be in charge of putting that at the top of `index.js`, and then somehow pass the instance somewhere?

Reply:
Amazing! I will try to give it another review tomorrow